cpp/src/lists/explode.cu

```diff
@@ -62,27 +62,35 @@ std::unique_ptr<table> build_table(
 
   std::vector<std::unique_ptr<column>> columns = gathered_table.release()->release();
 
-  auto inserted = columns.insert(columns.begin() + explode_column_idx,
-                                 explode_col_gather_map
-                                   ? std::move(detail::gather(table_view({sliced_child}),
-                                                              explode_col_gather_map->begin(),
-                                                              explode_col_gather_map->end(),
-                                                              cudf::out_of_bounds_policy::NULLIFY,
-                                                              stream,
-                                                              mr)
-                                                 ->release()[0])
-                                   : std::make_unique<column>(sliced_child, stream, mr));
+  columns.insert(columns.begin() + explode_column_idx,
+                 explode_col_gather_map
+                   ? std::move(detail::gather(table_view({sliced_child}),
+                                              explode_col_gather_map->begin(),
+                                              explode_col_gather_map->end(),
+                                              cudf::out_of_bounds_policy::NULLIFY,
+                                              stream,
+                                              mr)
+                                 ->release()[0])
+                   : std::make_unique<column>(sliced_child, stream, mr));
 
   if (position_array) {
     size_type position_size = position_array->size();
-    // the null mask for position matches the exploded column's gather map, so copy it over
-    rmm::device_buffer nullmask =
-      explode_col_gather_map ? copy_bitmask(*inserted->get()) : rmm::device_buffer(0, stream);
+    // build the null mask for position based on invalid entries in gather map
+    auto nullmask = explode_col_gather_map ? valid_if(
+                                               explode_col_gather_map->begin(),
+                                               explode_col_gather_map->end(),
+                                               [] __device__(auto i) { return i != -1; },
```

Comment:
Why do we have `-1` here? Is it some kind of "magic" number? If possible, I would prefer to have `i != InvalidIndex` or similar, where we declared `constexpr size_type InvalidIndex = -1;` somewhere else.

Reply:
Yes, this is magical and meshes with `cudf::out_of_bounds_policy::NULLIFY` on the gather. Adding `constexpr` to clean this up.

Reply:
Jake recommended to use `std::numeric_limits<size_type>::lowest()`, which seems to be better than `-1`. I'm not sure if it is relevant here.